test/src/test/java/jenkins/model/Jenkins56575Test.java

```diff
@@ -0,0 +1,227 @@
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+
+    private File jar;
+
+    private AtomicReference<Boolean> restartCalled = new AtomicReference<>(false);
+    private AtomicReference<Boolean> verifyRestartableCalled = new AtomicReference<>(false);
+    private Semaphore restartSemaphore = new Semaphore(0);
+
+    private Lifecycle previousLifecycle;
+    private Field instanceField;
+
+    @Before
+    public void setupMockLifecycle() throws Exception {
+        Main.isUnitTest = false;
+        j.jenkins.setCrumbIssuer(null);
+
+        previousLifecycle = Lifecycle.get();
+
+        instanceField = Lifecycle.class.getDeclaredField("INSTANCE");
+        instanceField.setAccessible(true);
+        instanceField.set(null, new Lifecycle() {
+            @Override
+            public void verifyRestartable() {
+                verifyRestartableCalled.set(true);
+            }
+
+            @Override
+            public void restart() {
+                restartCalled.set(true);
+                restartSemaphore.release();
+            }
+        });
+    }
+
+    @Before
+    public void grabCliJar() throws IOException {
+        jar = tmp.newFile("jenkins-cli.jar");
+        FileUtils.copyURLToFile(j.jenkins.getJnlpJars("jenkins-cli.jar").getURL(), jar);
+    }
+
+    @After
+    public void resetSettings() throws Exception {
+        Main.isUnitTest = true;
+
+        instanceField.set(null, previousLifecycle);
+    }
+
+    @Test
+    public void testRestart_regularClient() throws Exception {
+        WebClient wc = j.createWebClient()
+                .withRedirectEnabled(false)
+                .withThrowExceptionOnFailingStatusCode(false);
+        WebRequest request = new WebRequest(new URL(j.getURL() + "restart"), HttpMethod.POST);
+
+        assertThat(verifyRestartableCalled.get(), is(false));
+        assertThat(restartCalled.get(), is(false));
+
+        Page page = wc.getPage(request);
+        assertThat(page.getWebResponse().getStatusCode(), is(HttpURLConnection.HTTP_MOVED_TEMP));
+
+        restartSemaphore.acquire(1);
+
+        assertThat(verifyRestartableCalled.get(), is(true));
+        assertThat(restartCalled.get(), is(true));
+    }
+
+    @Test
+    @Issue("JENKINS-56575")
+    public void testRestartSimple_restClient() throws Exception {
+        WebClient wc = j.createWebClient()
+                .withRedirectEnabled(false)
+                .withThrowExceptionOnFailingStatusCode(false);
+        WebRequest request = new WebRequest(new URL(j.getURL() + "restartSimple"), HttpMethod.POST);
+
+        assertThat(verifyRestartableCalled.get(), is(false));
+        assertThat(restartCalled.get(), is(false));
+
+        // we cannot use wc.getPage with a HTTP_NO_CONTENT because the page is null, 
+        // compared to HTTP_ACCEPTED that is managed by WebClient by a blank page
+        WebResponse response = wc.loadWebResponse(request);
```

Comment:
Interesting point for the reviewer, the page is empty due to the 204 response code